### PR TITLE
Added weather platform card, with forecast graph

### PR DESCRIPTION
**Pull request by Tommatheussen**
This PR is related to the PR on the home-assistant main repo. (https://github.com/home-assistant/home-assistant/pull/4721). 

This PR in the main repo introduces forecast support on the weather platform (as well as implemented it in the openweathermap).
This PR focuses on showing the forecast data in a graph (inside a card), it still needs some work, since current data should be viewable as well, and the graph itself needs some work as well.

### CODE REVIEW

File: src/layouts/login-form.html [hunk 1]
```diff
@@ -50,7 +50,7 @@
       <img src="/static/icons/favicon-192x192.png" height="192" />
       <a href="#" id="hideKeyboardOnFocus"></a>
       <div class='interact'>
-        <div id='loginform' hidden$="[[showLoading]]">
+        <form id='loginform' hidden$="[[showLoading]]">
```
Comment (balloob) [inline src/layouts/login-form.html]: This is out of scope for this PR, however this would probably fix the saving password form dialog on Firefox, so I'm fine with leaving it in.

File: src/state-summary/state-card-weather.html [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          position: 'top'
+        },
+        interpolateNulls: true,
+        titlePosition: 'none',
+        vAxes: {
+          0: { title: "Celcius" },
+        },
+        legend: {
+          position: 'top',
+          title: 'Box Office Earnings in First Two Weeks of Opening',
```
Comment (balloob) [inline src/state-summary/state-card-weather.html]: 🤔

File: src/state-summary/state-card-weather.html [hunk 3]
```diff
@@ -0,0 +1,110 @@
+<link rel="import" href="../../bower_components/polymer/polymer.html">
+
+<link rel="import" href="../../bower_components/google-apis/google-legacy-loader.html">
+
+<dom-module id="state-card-weather">
+  <template>
```
Comment (balloob) [inline src/state-summary/state-card-weather.html]: We should at least include the name of the location

File: src/state-summary/state-card-weather.html [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+  attached: function () {
+    this.isAttached = true;
+  },
+
+  dataChanged: function() {
+    this.drawChart();
+  },
+
+  getDataArray: function() {
```
Comment (balloob) [inline src/state-summary/state-card-weather.html]: This function only cares about forecast instead of also including the current temperature.

File: src/state-summary/state-card-weather.html (hunk 3 above)
Comment (balloob) [inline src/state-summary/state-card-weather.html]: I also don't know if a chart is the best way to show the weather. We should put more emphasis on what is going on right now. I also think that probably instead of a `state-summary` we want to actually convert it to a stand alone card instead of showing it in line with the other data.

This example is obviously showing a lot of info but this is how we should approach showing the weather.

screen shot 2016-12-05 at 6 21 16 pm

File: src/layouts/login-form.html (hunk 1 above)
Comment (Tommatheussen) [inline src/layouts/login-form.html]: Damnit, forgot I made the PR from my master branch, I tested that change yesterday at home (after making it at work) and it didn't work, so needs some more work.
Ill remove this change later today.

File: src/state-summary/state-card-weather.html (hunk 3 above)
Comment (Tommatheussen) [inline src/state-summary/state-card-weather.html]: Yes indeed, I was planning on adding some more current information as well. Something like the shown example would be really nice (although I think OWM doesn't include all that information, like wind/precipitation forecast)

File: src/state-summary/state-card-weather.html (hunk 4 above)
Comment (Tommatheussen) [inline src/state-summary/state-card-weather.html]: You're right, it should probably start from the current time and temperature

File: src/cards/ha-weather-card.html [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+      },
+
+      stateObj: {
+        type: Object,
+        observer: 'drawChart'
+      },
+    },
+
+    computeCurrentWeather: function(stateObj) {
+      console.log(stateObj.attributes);
```
Comment (balloob) [inline src/cards/ha-weather-card.html]: stale log

File: src/cards/ha-weather-card.html [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+      return Object.keys(stateObj.attributes).filter(function(key) {
+        return ["attribution", "forecast", "friendly_name"].indexOf(key) === -1;
+      }).map(function(key, index) {
+        console.log(key, index);
```
Comment (balloob) [inline src/cards/ha-weather-card.html]: stale log

**Comment by Tommatheussen** [comment]
Question, when I changed this to be a separate card, does it still need to be listed in the `hass-util.html` and `ha-cards.html` files?

File: src/state-summary/state-card-content.html [hunk 7]
```diff
@@ -13,40 +13,40 @@
...
-      {
-        hass: hass,
-        stateObj: stateObj,
-        inDialog: inDialog,
-      });
-  },
-});
-</script>
+  });
+</script>
```
Comment (balloob) [inline src/state-summary/state-card-content.html]: missing new line

File: src/state-summary/state-card-content.html [hunk 8]
```diff
@@ -13,40 +13,40 @@
...
-
-  properties: {
-    hass: {
-      type: Object,
-    },
-
-    inDialog: {
-      type: Boolean,
-      value: false,
+  Polymer({
```
Comment (balloob) [inline src/state-summary/state-card-content.html]: All this indentation messes up the diff, making it difficult to see what has changed.

File: src/util/hass-util.html [hunk 9]
```diff
@@ -20,6 +20,7 @@
   'scene',
   'script',
   'weblink',
+  'weather',
```
Comment (balloob) [inline src/util/hass-util.html]: This should no longer be here as it is not the case.

File: src/state-summary/state-card-content.html (hunk 8 above)
Comment (Tommatheussen) [inline src/state-summary/state-card-content.html]: Indeed, seems my editor has wrong indentation set.
If I remember correctly there should be no changes in this file anymore, since at first I added `state-card-weather`, which is not used anymore.
I'll remove this asap.
